### PR TITLE
The massive Code Documentation PR

### PR DESCRIPTION
**Pull request by SamanthaDawsonBanks**
This will touch a *lot* of files so it will cause merge conflicts but it will be a WIP
to start with I will only be processing files that I have authored, bu that may expand

### CODE REVIEW

**Comment by SamanthaDawsonBanks** [comment]
P.S. there are a few refactors mixed in just to make this branch even harder to merge, but I spotted the issues during documenting the methods so they went in as well

File: mcomp_Java/src/common/datatypes/map/griddedMap/GriddedMap.java [hunk 1]
```diff
@@ -1,14 +1,21 @@
-/**
- * 
- */
 package common.datatypes.map.griddedMap;
 
 import common.datatypes.Waypoint;
 import common.datatypes.map.Map;
 
 /**
+ * The Top Level of the Girded / Amalgamated Map data structure.
```
Comment (HarryWJackson1996) [inline mcomp_Java/src/common/datatypes/map/griddedMap/GriddedMap.java]: spelling error "Girded"

**Review by HarryWJackson1996 — APPROVED** [review]
The Code Documentation is really clear. Its a great format to follow when documenting current and new code.

File: mcomp_Java/src/common/datatypes/map/griddedMap/GriddedMap.java (hunk 1 above)
Comment (SamanthaDawsonBanks) [reply mcomp_Java/src/common/datatypes/map/griddedMap/GriddedMap.java]: good catch ty mate

**Comment by SamanthaDawsonBanks** [comment]
this on is starting to diverge so it will need to be merged then done as a part 2

**Review by stevempope — APPROVED** [review]

**Pull request merged**